rmf_traffic_ros2/src/rmf_traffic_ros2/schedule/Node.cpp

```diff
@@ -633,34 +833,40 @@ void ScheduleNode::broadcast_participants()
 }
 
 //==============================================================================
-void ScheduleNode::request_changes(const RequestChanges& request)
+void ScheduleNode::request_changes(
+  [[maybe_unused]] const request_id_ptr& request_header,
+  const RequestChanges::Request::SharedPtr& request,
+  const RequestChanges::Response::SharedPtr& response)
 {
-  const auto query_topic = mirror_update_topics.find(request.query_id);
+  const auto query_topic = mirror_update_topics.find(request->query_id);
   if (query_topic == mirror_update_topics.end())
   {
     // Missing query update topic; something has gone very wrong.
     RCLCPP_ERROR(
       get_logger(),
       "[ScheduleNode::request_changes] Could not find mirror update topic "
-      "for query ID " + std::to_string(request.query_id));
+      "for query ID " + std::to_string(request->query_id));
+    response->result = RequestChanges::Response::UNKNOWN_QUERY_ID;
   }
   else
   {
     auto& mirror_update_topic_info = query_topic->second;
     // Tell the next update to send the changes since the requested version by
     // resetting the last sent version number to the requested version,
     // which may be std::nullopt if a full update is requested
-    if (request.full_update)
+    if (request->full_update)
     {
       mirror_update_topic_info.last_sent_version = std::nullopt;
     }
     else
     {
-      mirror_update_topic_info.last_sent_version = request.version;
+      mirror_update_topic_info.last_sent_version = request->version;
```

Comment:
If I'm reading this correctly, it's possible that `mirror_update_topic_info.last_sent_version` could have been set previously (e.g. by a different request), and this would be overwriting it.

I wonder if we should actually do

```
auto& last_version = mirror_update_topic_info.last_sent_version;
if (last_version.has_value() && rmf_utils::modular(request->version).less_than(*last_version))
  last_version = request->version;
```

That way we prioritize getting the biggest laggers up to date.

Reply:
Done in 770970e.